File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamLienQuan/SanPhamLienQuanColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamLienQuan")]
[BasedOnRow(typeof(SanPhamLienQuanRow), CheckNames = true)]
public class SanPhamLienQuanColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    public string SanPhamLienQuanTenSp { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamLienQuanColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamLienQuanRow } from "./SanPhamLienQuanRow";

export interface SanPhamLienQuanColumns {
    Id: Column<SanPhamLienQuanRow>;
    SanPhamTenSp: Column<SanPhamLienQuanRow>;
    SanPhamLienQuanTenSp: Column<SanPhamLienQuanRow>;
}

export class SanPhamLienQuanColumns extends ColumnsBase<SanPhamLienQuanRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamLienQuan';
    static readonly Fields = fieldsProxy<SanPhamLienQuanColumns>();
}